api/solarperformanceinsight_api/models.py

```diff
@@ -203,7 +204,21 @@ class PVsystModuleParameters(PVLibBase):
         gt=0,
     )
     _modelchain_dc_model: str = PrivateAttr("pvsyst")
-    _gamma: Optional[float] = PrivateAttr(None)
+    _gamma: float = PrivateAttr()
+
+    @root_validator(skip_on_failure=True)
+    def validate_diode_params(cls, values):
+        # and set _gamma here since it could raise an error with bad params
+        try:
+            pvlib.pvsystem.calcparams_pvsyst(
+                effective_irradiance=1000, temp_cell=25, **values
+            )
+            cls._gamma = pvsyst_temperature_coeff(**values)
+        except Exception:
```

Comment:
I don't think it is particularly helpful because I think the full traceback in a json object is too much and just the message of "float division by zero" isn't super helpful.